javaslang/src-gen/main/java/javaslang/API.java

```diff
@@ -84,6 +90,865 @@ private API() {
+     * @param methodReference A method reference
+     * @return A {@code Function2}
+     */
+    public static <T1, T2, R> Function2<T1, T2, R> f2(Function2<T1, T2, R> methodReference) {
+        return Function2.of(methodReference);
+    }
+
+    /**
+     * Alias for {@link Function3#of(Function3)}
+     *
+     * @param <R>             return type
+     * @param <T1>            type of the 1st argument
+     * @param <T2>            type of the 2nd argument
+     * @param <T3>            type of the 3rd argument
+     * @param methodReference A method reference
+     * @return A {@code Function3}
+     */
+    public static <T1, T2, T3, R> Function3<T1, T2, T3, R> f3(Function3<T1, T2, T3, R> methodReference) {
+        return Function3.of(methodReference);
+    }
+
+    /**
+     * Alias for {@link Function4#of(Function4)}
+     *
+     * @param <R>             return type
+     * @param <T1>            type of the 1st argument
+     * @param <T2>            type of the 2nd argument
+     * @param <T3>            type of the 3rd argument
+     * @param <T4>            type of the 4th argument
+     * @param methodReference A method reference
+     * @return A {@code Function4}
+     */
+    public static <T1, T2, T3, T4, R> Function4<T1, T2, T3, T4, R> f4(Function4<T1, T2, T3, T4, R> methodReference) {
+        return Function4.of(methodReference);
+    }
+
+    /**
+     * Alias for {@link Function5#of(Function5)}
+     *
+     * @param <R>             return type
+     * @param <T1>            type of the 1st argument
+     * @param <T2>            type of the 2nd argument
+     * @param <T3>            type of the 3rd argument
+     * @param <T4>            type of the 4th argument
+     * @param <T5>            type of the 5th argument
+     * @param methodReference A method reference
+     * @return A {@code Function5}
+     */
+    public static <T1, T2, T3, T4, T5, R> Function5<T1, T2, T3, T4, T5, R> f5(Function5<T1, T2, T3, T4, T5, R> methodReference) {
+        return Function5.of(methodReference);
+    }
+
+    /**
+     * Alias for {@link Function6#of(Function6)}
+     *
+     * @param <R>             return type
+     * @param <T1>            type of the 1st argument
+     * @param <T2>            type of the 2nd argument
+     * @param <T3>            type of the 3rd argument
+     * @param <T4>            type of the 4th argument
+     * @param <T5>            type of the 5th argument
+     * @param <T6>            type of the 6th argument
+     * @param methodReference A method reference
+     * @return A {@code Function6}
+     */
+    public static <T1, T2, T3, T4, T5, T6, R> Function6<T1, T2, T3, T4, T5, T6, R> f6(Function6<T1, T2, T3, T4, T5, T6, R> methodReference) {
+        return Function6.of(methodReference);
+    }
+
+    /**
+     * Alias for {@link Function7#of(Function7)}
+     *
+     * @param <R>             return type
+     * @param <T1>            type of the 1st argument
```

Comment:
I agree that it would increase readability / would turn focus on non-trivial things. But if we want formally ensure that no documentation is missing, we need to enable Java's doclint option.

Btw - I'm a little bit proud that we use doclint. Not everyone is able to do it because tons of Javadoc are broken and they are not able to effort the time to fix it.